### DISCUSSION TITLE
Markdown rending at Hover widget isn't the same

### DISCUSSION THREAD

**Discussion by yoavmil** [microsoft/vscode-discussions#1233] [Extension Development QnA] [answered]
Markdown rending at Hover widget isn't the same as regular Markdown rendering.
It lacks some markdown features like `math` and `mermaid`.

I've made an [extension](https://marketplace.visualstudio.com/items?itemName=YoavMiller.bamakom) (my first) that renders markdown inside a hover widget to allow developers to display picuters and diagram inside the code, where it's needed.

My question is why is the rendering missing features, and can I do to overcome?

**Comment by tamuratak** [comment, marked as answer]
> why is the rendering missing features

For security reasons. To reduce the attack surface. The markdown previewer is implemented in the sandbox using iframes. The hover widget is not.

> can I do to overcome?

You have to do markdown rendering on the extension host's side.

**Comment by Karliz24** [comment]
> > why is the rendering missing features
> 
> 
> 
> For security reasons. To reduce the attack surface. The markdown previewer is implemented in the sandbox using iframes. The hover widget is not.
> 
> 
> 
> > can I do to overcome?
> 
> 
> 
> You have to do markdown rendering on the extension host's side.
> 
> 
> 
>
